### PR TITLE
Remove `@add_reactions`

### PR DESCRIPTION
**Pull request by stelmo**
This is literally the only macro construction function. It does not fit in... 🪓

### CODE REVIEW

File: test/reconstruction/gapfill_minimum_reactions.jl [hunk 1]
```diff
@@ -6,20 +6,24 @@
...
-        "r8", m7 → m8, 0, 100
-        "r9", m8 → nothing, 0, 100
-        # "r10", m6 → nothing, 0, 100
-        "r11", m2 + m3 + m7 → nothing, 0, 100
-        "r12", m3 → m5, -10, 10
-    end
+    add_reactions!(
+        model,
+        [
+            # Reaction("r1", Dict("m1" => 1), :forward; upper_bound = 1),
```
Comment (exaexa) [inline test/reconstruction/gapfill_minimum_reactions.jl:12]: rm comment

File: test/reconstruction/gecko.jl [hunk 2]
```diff
@@ -78,14 +78,17 @@ end
...
-    end
+    add_reactions!(
+        model,
+        [
+            Reaction("r1"; Dict("m1" => 1), :forward),
+            Reaction("r2"; Dict("m2" => 1), :forward),
+            Reaction("r3"; Dict("m1" => -1, "m2" => -1, "m3" => 1), :forward),
+            Reaction("r4"; Dict("m3" => -1, "m4" => 1), :forward),
+            Reaction("r5"; Dict("m2" => -1, "m4" => -1), :bidirectional),
+            Reaction("r6"; Dict("m4" => -1), :forward),
```
Comment (exaexa) [inline test/reconstruction/gecko.jl:89]: the semicolons here are wrong (you'd need the name the next args)

**Review by exaexa — CHANGES_REQUESTED** [review]
basic testfixing, otherwise all ok

File: test/reconstruction/gapfill_minimum_reactions.jl (hunk 1 above)
Comment (stelmo) [reply test/reconstruction/gapfill_minimum_reactions.jl:12]: I kept the commented ones to make it obvious what is actually being tested (the same reactions later get added back. I think this makes it clear what is expected. I put a comment in the code for future devs

File: test/reconstruction/gapfill_minimum_reactions.jl [hunk 3]
```diff
@@ -6,20 +6,29 @@
...
+            # Reaction("r7", Dict("m2" => -1, "m7" => 1, "m6" => 1), :forward),
+            Reaction("r8", Dict("m7" => -1, "m8" => 1), :forward),
+            Reaction("r9", Dict("m8" => -1), :forward),
+            # Reaction("r10", Dict("m6" => -1), :forward),
+            Reaction("r11", Dict("m2" => -1, "m3" => -1, "m7" => -1), :forward),
+            Reaction(
+                "r12",
+                Dict("m3" => -1, "m5" => 1),
+                :forward;
+                lower_bound = -10,
```
[the remark this answers is not in the document]
Comment (stelmo) [reply test/reconstruction/gapfill_minimum_reactions.jl:27]: yeah, I am putting this on the roadmap

File: test/reconstruction/gapfill_minimum_reactions.jl (hunk 1 above)
Comment (exaexa) [reply test/reconstruction/gapfill_minimum_reactions.jl]: ok cool

**Review by exaexa — APPROVED** [review]

**Pull request merged**